### PR TITLE
Added Groovy and Kotlin versions of docs-examples

### PR DESCRIPTION
**Pull request by burtbeckwith**

### CODE REVIEW

File: oraclecloud-sdk/src/test/java/io/micronaut/oraclecloud/objectstorage/ObjectStorageClientTest.java [hunk 1]
```diff
@@ -1,85 +0,0 @@
...
-import org.junit.jupiter.api.Test;
-
-import java.util.List;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
-
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-
-@MicronautTest
-public class ObjectStorageClientTest {
```
Comment (graemerocher) [inline oraclecloud-sdk/src/test/java/io/micronaut/oraclecloud/objectstorage/ObjectStorageClientTest.java:21]: Why was this deleted?

File: oraclecloud-sdk/src/test/java/io/micronaut/oraclecloud/objectstorage/SdkTest.java [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-import com.oracle.bmc.functions.FunctionsInvokeAsyncClient;
-import com.oracle.bmc.functions.FunctionsInvokeClient;
-import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
-import org.junit.jupiter.api.Test;
-
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-
-// test that each SDK client is setup and injectable
-@MicronautTest
-public class SdkTest {
```
Comment (graemerocher) [inline oraclecloud-sdk/src/test/java/io/micronaut/oraclecloud/objectstorage/SdkTest.java:20]: Why was this deleted?

File: docs-examples/example-function-groovy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# tag::dockerfile[]
+FROM oracle/graalvm-ce:20.1.0-java11 as graalvm
```
Comment (graemerocher) [inline docs-examples/example-function-groovy/Dockerfile:2]: The docker files shouldn't be needed anymore apart from this uses a very old version of native image.

See https://micronaut.io/launch?javaVersion=JDK_11&lang=JAVA&build=GRADLE&test=JUNIT&name=demo&package=com.example&type=FUNCTION&features=oracle-function&version=2.5.6&activity=preview&showing=/build.gradle for an example build without the need for a `Dockerfile`

File: docs-examples/example-function-groovy/func.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+schema_version: 20180708
```
Comment (graemerocher) [inline docs-examples/example-function-groovy/func.yml:1]: This file is not needed either. You can deploy with the project fn CLI

File: docs-examples/example-function-kotlin/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# tag::dockerfile[]
+FROM oracle/graalvm-ce:20.1.0-java11 as graalvm
```
Comment (graemerocher) [inline docs-examples/example-function-kotlin/Dockerfile:2]: Same with this Dockerfile

**Review by graemerocher — CHANGES_REQUESTED** [review]

**Comment by burtbeckwith** [comment]
@graemerocher any idea why :docs-examples:example-java:testNativeImage fails (the mocks aren't injected) but the regular tests pass? everything else looks ok

**Comment by graemerocher** [comment]
Which tests? Where are the reports / logs?

**Comment by burtbeckwith** [comment]
https://github.com/micronaut-projects/micronaut-oracle-cloud/pull/211/checks?check_run_id=2876108913

**Comment by graemerocher** [comment]
probably due to the way the `testNativeImage` task works which is to spin up the native image as a separate process (and hence create a separate application context that doesn't see the mock) 

I would suggest to simply not run this test for native or change the assertions to just check the result is not empty

**Comment by burtbeckwith** [comment]
fixes https://github.com/micronaut-projects/micronaut-oracle-cloud/issues/214
